weaknesses/MASVS-CODE/MASWE-0076.md

```diff
@@ -1,38 +1,79 @@
-- **Non-Compliance**: Using outdated or vulnerable dependencies may result in noncompliance with security standards and regulations. This can expose businesses to regulatory penalties and legal liabilities, especially those in industries that handle sensitive data, such as healthcare or finance.
+- **Sensitive Data Exposure**: Vulnerable dependencies may be exploited to bypass access controls or cryptographic protections, which could lead to the exposure of sensitive user data, including credentials, session tokens, and personally identifiable information (PII). This can result in data breaches, which can have legal, financial and reputational consequences.
+- **Execution of Unauthorized Code or Privilege Escalation**: Exploitable flaws in embedded dependencies can allow attackers to execute arbitrary code within the app's context (e.g., through code injection or remote code execution (RCE)), escalate privileges, or manipulate app behavior. The overall impact can range from full compromise of user accounts, abuse of backend services or persistent access to protected resources. The consequences for the business can be severe, including financial loss, service disruption, and damage to customer trust.
+- **Regulatory and Policy Non-Compliance**: Including dependencies with publicly known CVEs may violate regulatory requirements (e.g., GDPR, HIPAA, PCI-DSS) or platform security policies (e.g., Google Play or App Store guidelines). Failure to update or remediate such vulnerabilities can result in app rejection, fines, or mandatory disclosures.
 
 ## Modes of Introduction
 
-Mobile apps rely heavily on dependencies. These dependencies can be "closed-source" through vendor products or "open-source" and maintained by the community.
-
-Dependencies can be implemented manually by adding them into the project and linking them, but they are usually added through dependency managers, which handle the integration into the project's files.
+- **Direct Dependencies**: Vulnerable dependencies can be introduced into the app either manually (by copying and linking source or binary files) or more commonly via package managers and build tools (e.g., Gradle, CocoaPods, Swift Package Manager). This includes both first- and third-party SDKs, and may involve both statically and dynamically linked libraries.
+- **Transitive Dependencies**: Dependencies can be pulled in indirectly through other libraries or SDKs that the app uses. This means that an app may still be affected by a vulnerable library if one of its dependencies includes it, even if the app does not directly include the library.
+- **Dynamically Loaded Dependencies**: Some libraries may be dynamically loaded at runtime, which can make it difficult to track and manage dependencies. This can lead to situations where a vulnerable version of a library is used without the developer's knowledge.
```

Comment:
```suggestion
- **Dynamically Loaded Dependencies**: Some libraries may be dynamically loaded at runtime, which can make it difficult to track and manage dependencies. This can lead to situations where a vulnerable version of a library is used without the developer's knowledge.
```
This goes against Google Developer policy (dynamically loading external code), unless maybe for JS libraries. We could keep it, but it's a bit unlikely.